### DISCUSSION TITLE
T-Display S3 : Simple Script to turn Screen on an off

### DISCUSSION THREAD

**Discussion by hammikb** [Bodmer/TFT_eSPI#2328] [Q&A - General]
When I was first getting into the T-display S3 and ESP32 devices in general I wish more people would post what they found when they fixed issues, so we would have documentation and just tricks in general. So below is a simple script to turn off the Backlight and save same energy. 

This is all based on a 
- T-Display-S3 
- Arduino IDE
- TFT_eSPI 2.5.0

```
//This just needs to be called before everything else below 

bool ledState = LOW;                       //Current State of PIN
unsigned long lastPress = 0;            //Time since last pressed 
const int debounceDelay = 50;        //Time between button pushes for it to register 

//Function to check for the button being pushed 
void powerswitch() {
  int buttonState = digitalRead(0);      // 0 can change to whatever GPIO you are using 
  if (buttonState == LOW && millis() - lastPress > debounceDelay) {
    ledState = !ledState;                       // toggle the current state of the LED
    digitalWrite(38, ledState);              // 38 can change to whatever GPIO controls the BL (backlight)
    lastPress = millis();
  }
}

//Calling the function in loop()
void loop() {
  powerswitch();
}
```

This script by no means turns off the battery or stops the flow of power it just reduces it

**Comment by Bodmer** [comment]
Thanks for sharing.

**Reply by brentfraser** [reply]
Interesting stuff.  I'm using my T-Display-S3 (with an external DAC module) to render audio to the AUX in of my stereo.  It works great but I'm concerned about screen burn-in since it will be on most of the time.  Will turning off the backlight be sufficient to prevent burn in?

Thanks!

**Reply by hammikb** [reply]
I'm not exactly 100% sure but I would say yes turning it off would prevent burning of the image or burning out the screen. You can also can control the brightness of screen , if I find the script I'll send it here

**Reply by redeyedalice** [reply]
Turning off the backlight may help prevent burn in as it will lower the heat and overall energy present. However the pixels are still being powered. Burn in happens when a pixel can no longer return to their off state.  In theory having the pixels powered on even when the backlight is off may still cause lcd burn in. I've never tested a T-Display S3 long enough to know how long it takes for burn in.

The below will power off the screen completely for the T-Display S3 if you are using TFT_eSPI. 
      tft.writecommand(ST7789_DISPOFF); //turn off lcd display
      digitalWrite(38, LOW); //turn of lcd backlight

When you want to turn it back on you can use the below. In my experience you need to reinitialize the screen.
      tft.init();
      tft.writecommand(ST7789_DISPON); //turn on display
      digitalWrite(38, HIGH); //turn on lcd backlight

**Reply by brentfraser** [reply]
Thanks I'll give it a try! I was wondering if I needed to turn the display off.  I'll need to do some re-org of my code managing writing info to the display, but it should be re-written anyway to manage screens of different resolutions (especially if I do buy a t-display-amoled)

**Reply by phillmj** [reply]
Interesting, I had developed a similar script, and I set the BL to 0 with a digitalWrite. To turn it back on, I just write the previous backlight value to the BL pin, and it lights up again no problem. No tft.init needed. It was a different screen I'm sure but I'm surprised you have to reinitialize.

**Reply by redeyedalice** [reply]
@brentfraser 
While I've not used the amoled yet myself, it is not just a different size but has a rm67162 and the commands seem to be different as it isn't included in TFT_eSPI. Looking at rm67162.cpp/h in the LilyGo examples there is lcd_sleep(); that should do what you are looking for if you move to the amoled. 

@phillmj  
If you just turn off the backlight but not the display controller then you do not need to reinitialize the screen. However ST7789_DISPOFF turns the entire unit off. If you didn't want to have to use tft.init() or needed to maintain the last screen drawn you could try commands ST7789_SLPIN/ST7789_SLPOUT instead of ST7789_DISPOFF/ST7789_DISPON. This would put the tft into sleep mode.

**Comment by git2212** [comment]
There is an AMOLED version of this puppy now [https://www.lilygo.cc/products/t-display-s3-amoled](https://www.lilygo.cc/products/t-display-s3-amoled). Theoretically a modern AMOLED should be a lot less susceptible to burn-in, especially if driven at lower intensity No?
